ibis/pandas/aggcontext.py

```diff
@@ -405,17 +409,71 @@ def sliced_agg(s):
 
         # perform the per-group rolling operation
         windowed = self.construct_window(grouped)
-        with warnings.catch_warnings():
-            warnings.filterwarnings(
-                "ignore", message=".+raw=True.+", category=FutureWarning
+
+        if callable(function):
+            # Use custom logic to computing rolling window UDF instead of
+            # using pandas's rolling function.
+            # This is because pandas's rolling function doesn't support
+            # multi param UDFs.
+            def create_input_gen(grouped_series, window_size):
+                # create a generator for each input series
+                # the generator will yield a slice of the
+                # input series for each valid window
+                data = getattr(grouped_series, 'obj', grouped_series).values
```

Comment:
I choose to do ndarray here because:

- This preserve the same udf API as before. 
- Passing Series is about 15x slower than ndarray

I don't want to introduce API change and performance regression in this PR. I think we can have a separate chat whether window UDF should take ndarray or Series.

Reply:
you shouldn’t show any perf issues must be something odd going on

does it currently take ndarray?

Reply:
Yes it currently take ndarray (By using raw=True I think)

Reply:
ok can you create an issue to fix this, meaning to use iloc. .values is not type preserving and generally a bad idea.

Reply:
Created: https://github.com/ibis-project/ibis/issues/2045